### PR TITLE
DOM#6

### PR DESCRIPTION
**Pull request by CrazyPike**
### DOM

[Demo](
https://crazypike.github.io/frontend-2019-homeworks/submissions/CrazyPike/DOM%236/sidemenu.html) |
[Code base](https://github.com/CrazyPike/frontend-2019-homeworks/tree/master/submissions/CrazyPike/DOM%236)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by A-Ostrovnyy** [comment]
Your demo link didn't work. Please fix.

**Comment by CrazyPike** [comment]
fixed

File: submissions/CrazyPike/DOM#6/sidemenu.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+    <link href="style.css" rel="stylesheet">
+</head>
+<body>
+<div id="menu"></div><div id="main"></div>
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/sidemenu.html]: Use wrapper for  easier control your grid.

File: submissions/CrazyPike/DOM#6/sidemenu.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/sidemenu.html]: Add `meta` tag for mobile screen device.

File: submissions/CrazyPike/DOM#6/style.css [hunk 3]
```diff
@@ -0,0 +1,33 @@
+body{
+    display:flex;
+}
+#menu{
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/style.css]: Avoid styling by `#id` and tag name.
For `<body>` leave but for other tags it is better to assign them a class and style them according to the class.
When a project is large and the team is working on it, it avoids the constant dropping of styles for the tag in different places. In addition, the class can be applied to different tags.

File: submissions/CrazyPike/DOM#6/style.css [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+}
+#main{
+    width: 70%;
+    border: 1px solid lightgrey;
+    border-left: none;
+    background-color: rgba(231,231,231,0.6);
+}
+.itemMenu{
+    border-top: 1px solid lightgrey;
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/style.css]: In this case better set all borders and in the next line remove `border-bottom` like in the line 12 - 13.

File: submissions/CrazyPike/DOM#6/index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+elementMenu.innerHTML = menuItems.map((item,index) => `<div class="itemMenu" id="item${index}">${item.name}</div>`).join(' ');
+elementMenu.addEventListener('click', (event) =>{
+    let target = event.target;
+    if (!(target.className !== 'itemMenu' || target.className !== 'itemMenu active')) return;
+    document.getElementById(`item${previousItem}`).className = 'itemMenu';
+    target.className = 'itemMenu active';
+    previousItem = target.id.match(/\d+/g)[0];
+    document.getElementById('main').innerText = menuItems[target.id.match(/\d+/g)[0]]['text'];
+});
+document.getElementById('item0').click();
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/index.js]: Add empty line in the end of every file.

File: submissions/CrazyPike/DOM#6/sidemenu.html (hunk 1 above)
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/sidemenu.html]: For menu better use `<nav>` tag and for main content - `<main>`.
Make your markup more semantic.

File: submissions/CrazyPike/DOM#6/index.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+let previousItem = 0;
+let menuItems = [{name:'topic1', text: 'text1'},
+    {name:'topic2', text: 'text2'},
+    {name:'topic3', text: 'text3'},
+    {name:'topic4', text:   'text4'}];
+let elementMenu = document.getElementById('menu');
+elementMenu.innerHTML = menuItems.map((item,index) => `<div class="itemMenu" id="item${index}">${item.name}</div>`).join(' ');
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/index.js]: Too much logic in one line. Separate it on logical blocks for more readable code. 
Example: generate menu list and next line - include menu list inside menu wrapper.

File: submissions/CrazyPike/DOM#6/index.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+elementMenu.addEventListener('click', (event) =>{
+    let target = event.target;
+    if (!(target.className !== 'itemMenu' || target.className !== 'itemMenu active')) return;
+    document.getElementById(`item${previousItem}`).className = 'itemMenu';
+    target.className = 'itemMenu active';
+    previousItem = target.id.match(/\d+/g)[0];
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/index.js]: Easier will be add `data-attribute` to list item and then onclick get this value and use it as index of your array with data.

File: submissions/CrazyPike/DOM#6/index.js [hunk 8]
```diff
@@ -4,13 +4,14 @@ let menuItems = [{name:'topic1', text: 'text1'},
     {name:'topic3', text: 'text3'},
     {name:'topic4', text:   'text4'}];
 let elementMenu = document.getElementById('menu');
-elementMenu.innerHTML = menuItems.map((item,index) => `<div class="itemMenu" id="item${index}">${item.name}</div>`).join(' ');
+let menuList = menuItems.map((item,index) => `<li class="itemMenu" id="item${index}" itemNumber=${index}>${item.name}</li>`).join(' ');
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/index.js]: When you start using attribute `itemNumber` you don't need `id="itemN"`.

File: submissions/CrazyPike/DOM#6/sidemenu.html [hunk 9]
```diff
@@ -2,11 +2,18 @@
 <html lang="en">
 <head>
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>Title</title>
```
Comment (A-Ostrovnyy) [inline submissions/CrazyPike/DOM#6/sidemenu.html]: Change Title to more appropriate.

File: submissions/CrazyPike/DOM#6/index.js (hunk 8 above)
Comment (CrazyPike) [reply submissions/CrazyPike/DOM#6/index.js]: removed ids, reworked the code with attribute 'item'

**Comment by A-Ostrovnyy** [comment]
Good job!

**Pull request merged**